test/with_dummyserver/test_socketlevel.py

```diff
@@ -1225,6 +1225,38 @@ def socket_handler(listener):
         pool.request('GET', '/', headers=OrderedDict(expected_request_headers))
         self.assertEqual(expected_request_headers, actual_request_headers)
 
+    def test_request_host_header_ignores_fqdn_dot(self):
+
+        received_host = []
+
+        def socket_handler(listener):
+            sock = listener.accept()[0]
+
+            buf = b''
+            while not buf.endswith(b'\r\n\r\n'):
+                buf += sock.recv(65536)
+
+            headers_list = [header for header in buf.split(b'\r\n')[1:] if header]
+
+            for header in headers_list:
+                (key, value) = header.split(b': ')
+                if key.decode('ascii').startswith(u'Host'):
+                    received_host.append(value.decode('ascii'))
+
```

Comment:
Might be nice to pull this header parsing logic out into a separate helper function. It's the kind of thing that might come up from time to time.

Reply:
Makes sense! Something like this?

```python
def assert_header_received(self, received_headers, header_name, expected_value=None):
    """
    Asserts that a header with the given name was received; if a value is passed to
    `expected_value`, asserts that the header has that value. 
    """
    pass
```

Reply:
Yeah, sounds good to me!